x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptFieldType.java

```diff
@@ -239,6 +235,10 @@ protected final void doXContentBody(XContentBuilder builder, boolean includeDefa
 
         protected abstract AbstractScriptFieldType<?> buildFieldType();
 
+        protected final Script defaultScript() {
+            return new Script("emitValues(\"" + name + "\")");
```

Comment:
Just a though, a bit controversial I reckon: although we are not documenting emitValues, I wonder if it's worth iterating on it and making it closer to what we will need in the long run. I see that the function is context dependent in how it extracts values from source. Would it make sense to instead have parse* functions that are independent of the context and can potentially be used in any context? I would be ok even if the default script loops over the returned values.